src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts

```diff
@@ -287,223 +240,16 @@ export class NotebookContribution extends Disposable implements IWorkbenchContri
 		this._notebookDiffEditorIsOpen.set(this.editorService.activeEditorPane?.getId() === NOTEBOOK_DIFF_EDITOR_ID);
 	}
 
-	getUserAssociatedEditors(resource: URI) {
-		const rawAssociations = this.configurationService.getValue<EditorsAssociations>(editorsAssociationsSettingId) || [];
-
-		return coalesce(rawAssociations
-			.filter(association => CustomEditorInfo.selectorMatches(association, resource)));
-	}
-
 	getUserAssociatedNotebookEditors(resource: URI) {
-		const rawAssociations = this.configurationService.getValue<EditorsAssociations>(editorsAssociationsSettingId) || [];
+		const associationsForResource = this.extensionContributedEditorService.getAssociationsForResource(resource);
 
-		return coalesce(rawAssociations
-			.filter(association => CustomEditorInfo.selectorMatches(association, resource))
+		return coalesce(associationsForResource
 			.map(association => this.notebookService.getContributedNotebookProvider(association.viewType)));
 	}
```

Comment:
I believe there is no more user of this function and you can nuke it all together

Reply:
Removed